src/message/fetchActions.js

```diff
@@ -168,6 +171,17 @@ const initialFetchComplete = (): Action => ({
   type: INITIAL_FETCH_COMPLETE,
 });
 
+const initialFetchAbortPlain = (): Action => ({
+  type: INITIAL_FETCH_ABORT,
+});
+
+// This will be used in an upcoming commit.
+/* eslint-disable-next-line no-unused-vars */
+export const initialFetchAbort = () => async (dispatch: Dispatch, getState: GetState) => {
+  NavigationService.dispatch(resetToAccountPicker());
+  dispatch(initialFetchAbortPlain());
```

Comment:
Before #3804 was resolved, we would store React Navigation's state in our Redux store, and we had a `navReducer` that we could make recognize an action like `initialFetchAbortPlain()` and update the nav state accordingly.

(We actually mixed that approach with dispatching actions from the action creators in `navActions`, with the same `dispatch` we use for all the other actions. `NavigationService` is a stepping stone toward using React Navigation's [`navigation` prop/object](https://reactnavigation.org/docs/navigation-prop/).)

With React Navigation's state not in our Redux store anymore, that old approach in `navReducer` isn't available.

We also can't make a go-and-navigate side effect to our existing reducers because reducers [explicitly aren't the place for side effects](https://redux.js.org/tutorials/fundamentals/part-3-state-actions-reducers#rules-of-reducers).

The choice to put `NavigationService.dispatch(resetToAccountPicker());` (or I guess, one day, `navigation.reset({ index: 0, routes: [{ name: 'main-tabs' }] });` or similar) in the thunk action creator `initialFetchAbort` is a bit subtle: it follows our strategy to eliminate a class of bugs where we try to do stuff before the store has rehydrated; Greg has a good explanation of that [here](https://chat.zulip.org/#narrow/stream/48-mobile/topic/.60remoteNotificationsRegistered.60.20listener.20too.20late/near/1148871).

I've also thought it'd be nice to not count on remembering to call go-navigate code alongside dispatching things like `logout()`. The fact that we do is a reminder that dispatching `logout()` doesn't…finish the job of logging out: navigating away from the logged-in screens is an important part of that. 🙂

[`store.subscribe()`](https://redux.js.org/api/store#subscribelistener) comes to mind, but that's a low-level API and it doesn't tell a passed handler about actions that have been dispatched; only the fact that an action has been dispatched.

I think that leaves Redux [middleware](https://redux.js.org/understanding/history-and-design/middleware)—I could imagine writing some middleware that does things like see an `INITIAL_FETCH_ABORT`-type action go by, and navigate based on that.

Reply:
> I've also thought it might be nice to not count on remembering to call go-navigate-somewhere code alongside dispatching things like `logoutPlain()`. The fact that we do is a reminder that dispatching `logoutPlain()` doesn't…finish the job of logging out: navigating away from the logged-in screens is an important part of that. 🙂

Ah, I just edited `logout()` here to `logoutPlain()`, after I saw that that's what I meant. I think the situation there isn't quite as bad as I'd thought; `logout()` (the thunk action) does do the essential job of navigating. 🙂

Reply:
Similarly, the thunk action `initialFetchAbort()` does do the essential job of navigating, while the plain action `initialFetchAbortPlain()` doesn't.

Reply:
Let me know if I can unpack any of that, maybe on CZO. 🙂